### PR TITLE
Build

### PR DESCRIPTION
**Pull request by will-moore**
Same as https://github.com/ome/omero-gallery/pull/112/files
Needed to fix build at https://github.com/IDR/deployment/pull/441#pullrequestreview-2459464951 since the last release of idr-gallery didn't get pushed to pypi.
cc @jburel

### CODE REVIEW

File: .github/workflows/publish_pypi.yml [hunk 1]
```diff
@@ -7,12 +7,14 @@ jobs:
...
+          python-version: '3.10'
       - name: Build a binary wheel and a source tarball
         run: |
-          python -mpip install wheel
-          python setup.py sdist bdist_wheel
+          python -mpip install build
+          python -m build
       - name: Publish distribution to PyPI
         if: startsWith(github.ref, 'refs/tags')
         uses: pypa/gh-action-pypi-publish@v1.3.0
```
Comment (jburel) [inline .github/workflows/publish_pypi.yml:20]: Could you also bump that action to 1.8.14?

**Comment by jburel** [comment]
could you please bump ``actions/checkout`` in prettier workflow?

**Comment by will-moore** [comment]
Errors. Fixing now...
```
./idr_gallery/urls.py:24:80: E501 line too long (80 > 79 characters)
./idr_gallery/views.py:142:80: E501 line too long (80 > 79 characters)
./idr_gallery/views.py:165:80: E501 line too long (94 > 79 characters)
```

**Comment by will-moore** [comment]
Permissions issue!
```
+ rm -rf build dist idr_gallery.egg-info 'src/*egg-info'
OCI runtime exec failed: exec failed: unable to start container process: exec: "//idr-gallery/.omeroci/app-build": permission denied: unknown
```

**Comment by will-moore** [comment]
@jburel Looking better now...

**Review by jburel — APPROVED** [review]

**Pull request merged**